bin/database.py

```diff
@@ -1111,6 +1111,23 @@ def upgrade_to_30():
     cursor = config.db.projects.find({'files.name': {'$exists': True}, 'files.created': {'$exists': False}})
     process_cursor(cursor, upgrade_to_30_closure_coll, context = 'projects')
 
+def upgrade_to_31_closure(group):
+    result = config.db.groups.update_one({'_id': group['_id']}, {'$rename': {'roles': 'permissions'}})
+
+    if result.modified_count == 1:
+        return True
+    else:
+        return 'Parallel failed: update doc ' + str(job['_id']) + ' resulted modified ' + str(result.modified_count)
+
+def upgrade_to_31():
+    """
+    Changes group.roles -> groups.permissions
+
+    scitran/core #662
+    """
+    cursor = config.db.groups.find({'roles': {'$exists': True}})
```

Comment:
You don't always have to use a process_cursor, only when the change can't be done entirely in mongo (usually when something needs to be computed or files/analyses/other list object are involved).

Here you could do `config.db.groups.update_many({}, {'$rename': {'roles': 'permissions'}})`